common/launch_test.go

```diff
@@ -2,47 +2,206 @@ package common
+			},
+		},
+		"executablePath_err": {
+			opts: map[string]any{"executablePath": 1},
+			err:  "executablePath should be a string",
+		},
+		"headless": {
+			opts: map[string]any{
+				"headless": false,
+			},
+			assert: func(tb testing.TB, lo *LaunchOptions) {
+				tb.Helper()
+				assert.False(t, lo.Headless)
+			},
+		},
+		"headless_err": {
+			opts: map[string]any{"headless": "ABC"},
+			err:  "headless should be a boolean",
+		},
+		"ignoreDefaultArgs": {
+			opts: map[string]any{
+				"ignoreDefaultArgs": []string{"--hide-scrollbars", "--hide-something"},
+			},
+			assert: func(tb testing.TB, lo *LaunchOptions) {
+				tb.Helper()
+				assert.Len(t, lo.IgnoreDefaultArgs, 2)
+				assert.Equal(t, "--hide-scrollbars", lo.IgnoreDefaultArgs[0])
+				assert.Equal(t, "--hide-something", lo.IgnoreDefaultArgs[1])
+			},
+		},
+		"ignoreDefaultArgs_err": {
+			opts: map[string]any{"ignoreDefaultArgs": "ABC"},
+			err:  "ignoreDefaultArgs should be an array of strings",
+		},
+		"logCategoryFilter": {
+			opts: map[string]any{
+				"logCategoryFilter": "**",
+			},
+			assert: func(tb testing.TB, lo *LaunchOptions) {
+				tb.Helper()
+				assert.Equal(t, "**", lo.LogCategoryFilter)
+			},
+		},
+		"logCategoryFilter_err": {
+			opts: map[string]any{"logCategoryFilter": 1},
+			err:  "logCategoryFilter should be a string",
+		},
+		"proxy": {
+			opts: map[string]any{
+				"proxy": ProxyOptions{
+					Server:   "serverVal",
+					Bypass:   "bypassVal",
+					Username: "usernameVal",
+					Password: "passwordVal",
+				},
+			},
+			assert: func(tb testing.TB, lo *LaunchOptions) {
+				tb.Helper()
+				assert.Equal(t, ProxyOptions{
+					Server:   "serverVal",
+					Bypass:   "bypassVal",
+					Username: "usernameVal",
+					Password: "passwordVal",
+				}, lo.Proxy)
+			},
+		},
+		"proxy_err": {
+			opts: map[string]any{"proxy": 1},
+			err:  "proxy should be an object",
+		},
+		"slowMo": {
+			opts: map[string]any{
+				"slowMo": "5s",
+			},
+			assert: func(tb testing.TB, lo *LaunchOptions) {
+				tb.Helper()
+				assert.Equal(t, 5*time.Second, lo.SlowMo)
+			},
+		},
+		"slowMo_err": {
+			opts: map[string]any{"slowMo": "ABC"},
+			err:  "slowMo should be a time duration value",
+		},
+		"timeout": {
+			opts: map[string]any{
+				"timeout": "10s",
+			},
+			assert: func(tb testing.TB, lo *LaunchOptions) {
+				tb.Helper()
+				assert.Equal(t, 10*time.Second, lo.Timeout)
+			},
+		},
+		"timeout_err": {
+			opts: map[string]any{"timeout": "ABC"},
+			err:  "timeout should be a time duration value",
+		},
+	} {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
 			t.Parallel()
-
-			vu := k6test.NewVU(t)
-			opts := vu.ToGojaValue(tc.opts)
-			lopts := NewLaunchOptions()
-			err := lopts.Parse(vu.Context(), opts)
-			require.NoError(t, err)
-			tc.assert(t, lopts)
+			var (
+				vu = k6test.NewVU(t)
+				lo = NewLaunchOptions()
+			)
+			if err := lo.Parse(vu.Context(), vu.ToGojaValue(tt.opts)); tt.err != "" {
```

Comment:
I'd prefer:

```go
if tt.err != "" {
    err := lo.Parse(vu.Context(), vu.ToGojaValue(tt.opts))
    require.ErrorContains(t, err, tt.err)
}
```

I think that's easier to understand.

Reply:
I guess you mean the following:

```go
err := lo.Parse(vu.Context(), vu.ToGojaValue(tt.opts))
if tt.err != "" {
	require.ErrorContains(t, err, tt.err)
} else {
	require.NoError(t, err)
	tt.assert(t, lo)
}
```

Reply:
Resolved in 712930a.

Reply:
@ankur22, this is resolved in 2f9a61a.